src/api/StatusApi.ts

```diff
@@ -13,6 +13,60 @@ class StatusApi {
         return await this.statusRepo.getStatus()
     }
 
+    // checks all dates and create yellow and red alerts if too old
+    // returns an object with the alerts if there are any
+    getAgeAlerts(
+        statusData: {
+            [podcastId: number]: {
+                latestUpdates: { [endpointName: string]: Date }
+            }
+        },
+        yellowAgeHours: number,
+        redAgeHours: number
+    ) {
+        type AlertData = {
+            podcastId: number
+            endpoint: string
+            ageHours: number
+        }
+        const yellowAlerts: AlertData[] = []
+        const redAlerts: AlertData[] = []
+        Object.entries(statusData).forEach(([podcastIdStr, podcastData]) => {
+            // as entries returns key as string, cast to number
+            const podcastId = parseInt(podcastIdStr)
+            Object.entries(podcastData.latestUpdates).forEach(
+                ([endpointName, endpointDate]) => {
+                    const ageHours = Math.round(
+                        (new Date().getTime() -
+                            new Date(endpointDate).getTime()) /
+                            (1000 * 60 * 60)
+                    )
+                    if (ageHours >= redAgeHours) {
+                        redAlerts.push({
+                            podcastId: podcastId,
+                            endpoint: endpointName,
+                            ageHours: ageHours,
+                        })
+                    } else if (ageHours >= yellowAgeHours) {
+                        yellowAlerts.push({
+                            podcastId: podcastId,
+                            endpoint: endpointName,
+                            ageHours: ageHours,
+                        })
+                    }
+                }
+            )
+        })
+        const alerts: { yellow?: AlertData[]; red?: AlertData[] } = {}
+        if (yellowAlerts.length > 0) {
+            alerts.yellow = yellowAlerts
+        }
+        if (redAlerts.length > 0) {
+            alerts.red = redAlerts
+        }
+        return alerts
```

Comment:
Can't this be simplified to 

```suggestion
        return { yellow: yellowAlerts,  red: redAlerts };
```

because you already set up the `AlertData` arrays before?

Reply:
added comments to explain it:

        // only add alerts if there are any
        // this allows the client to look for the presence of the key yellow or red
        // to determine if there are any alerts. this is easier for monitors as
        // they don't have to parse json